include/deal.II/numerics/matrix_creator.templates.h

```diff
@@ -632,22 +632,56 @@ namespace MatrixCreator
 
 
     } // namespace AssemblerBoundary
+
+    
+
+    template <typename T>
+    using compress_t = decltype(std::declval<const T>().compress(
+      std::declval<	VectorOperation>()));
```

Comment:
Is `const T` what you want? I think this means that `compress()` must be a `const` member function for this to work (though I admit that I'm not entirely sure). Either way, it's probably easier to read without the `const`.